cmd/main.go

```diff
@@ -71,163 +100,233 @@ var (
 
-	// delete
-	contactDeleteCmd     = contactCmd.Command("delete", "Deletes a known contact").Alias("del").Alias("remove").Alias("rm")
-	contactDeleteAddress = contactDeleteCmd.Arg("address", "Account Address").Required().String()
-
-	// search
-	contactSearchCmd     = contactCmd.Command("search", "Searches locally and on the network for contacts").Alias("find")
-	contactSearchName    = contactSearchCmd.Flag("name", "Search by display name").Short('n').String()
-	contactSearchAddress = contactSearchCmd.Flag("address", "Search by account address").Short('a').String()
-	contactSearchLocal   = contactSearchCmd.Flag("only-local", "Only search local contacts").Bool()
-	contactSearchRemote  = contactSearchCmd.Flag("only-remote", "Only search remote contacts").Bool()
-	contactSearchLimit   = contactSearchCmd.Flag("limit", "Stops searching after [limit] results are found").Default("5").Int()
-	contactSearchWait    = contactSearchCmd.Flag("wait", "Stops searching after [wait] seconds have elapsed (max 30s)").Default("2").Int()
+	// contact search
+	contactSearchCmd     := contactCmd.Command("search", "Searches locally and on the network for contacts").Alias("find")
+	contactSearchName    := contactSearchCmd.Flag("name", "Search by display name").Short('n').String()
+	contactSearchAddress := contactSearchCmd.Flag("address", "Search by account address").Short('a').String()
+	// @todo what is the point of this, as `textile contact get <address>` can do this?
+	contactSearchLocal   := contactSearchCmd.Flag("only-local", "Only search local contacts").Bool()
+	contactSearchRemote  := contactSearchCmd.Flag("only-remote", "Only search remote contacts").Bool()
+	contactSearchLimit   := contactSearchCmd.Flag("limit", "Stops searching after [limit] results are found").Default("5").Int()
+	contactSearchWait    := contactSearchCmd.Flag("wait", "Stops searching after [wait] seconds have elapsed (max 30s)").Default("2").Int()
+	cmds[contactSearchCmd.FullCommand()] = func () error {
+		return ContactSearch(*contactSearchName, *contactSearchAddress, *contactSearchLocal, *contactSearchRemote, *contactSearchLimit, *contactSearchWait)
+	}
+	// @todo why not make this part of `textile contact list`?
+	// @todo perhaps our `list` and `get` commands can be merged
 
 	// ================================
 
 	// daemon
-	daemonCmd      = appCmd.Command("daemon", "Start a node daemon session")
-	daemonRepoPath = daemonCmd.Flag("repo-dir", "Specify a custom repository path").Short('r').String()
-	daemonPinCode  = daemonCmd.Flag("pin-code", "Specify the pin code for datastore encryption (omit no pin code was used during init)").Short('p').String()
-	daemonDocs     = daemonCmd.Flag("serve-docs", "Whether to serve the local REST API docs").Short('s').Bool()
-	// @note use global debug flag, as otherwise conflict arises
+	daemonCmd  := appCmd.Command("daemon", "Start a node daemon session")
+	daemonRepo := daemonCmd.Flag("repo", "Specify a custom repository path").Short('r').String()
```

Comment:
Can you update https://github.com/textileio/go-textile/blob/master/Dockerfile#L81 and https://github.com/textileio/go-textile/blob/master/Dockerfile.cafe#L86 ?